packages/discord.js-utilities/src/lib/PaginatedMessages/PaginatedMessage.ts

```diff
@@ -753,46 +807,106 @@ export class PaginatedMessage {
+			// If we are to emit a warning,
+			//   then check if the interaction is an interaction based command,
+			//   and check if the client has the `'CHANNEL'` partial,
+			//   in which case we don't want to emit a warning.
+			if (shouldEmitWarning && isInteraction && messageOrInteraction.client.options.partials?.includes('CHANNEL')) {
+				shouldEmitWarning = false;
+			}
 
-			// If a PaginatedMessage was found then stop it
-			paginatedMessage?.collector?.stop();
+			// IF we are to emit a warning,
+			//   then check if the interaction is a message based command,
+			//   and check if the client has the 'CHANNEL' partial,
+			//   and check if the client has the 'DIRECT_MESSAGE' intent',
+			//   in which case we don't want to emit a warning.
+			if (
+				shouldEmitWarning &&
+				!isInteraction &&
+				messageOrInteraction.client.options.partials?.includes('CHANNEL') &&
+				new Intents(messageOrInteraction.client.options.intents).has(Intents.FLAGS.DIRECT_MESSAGES)
+			) {
+				shouldEmitWarning = false;
+			}
 
-			// If the message was sent by a bot, then set the response as this one
-			if (runsOnInteraction(messageOrInteraction)) {
-				if (messageOrInteraction.user.bot && messageOrInteraction.user.id === messageOrInteraction.client.user?.id) {
-					this.response = messageOrInteraction;
-				}
-			} else if (messageOrInteraction.author.bot && messageOrInteraction.author.id === messageOrInteraction.client.user?.id) {
-				this.response = messageOrInteraction;
+			// If we should emit a warning then do so.
+			if (shouldEmitWarning) {
+				process.emitWarning(
+					[
+						'PaginatedMessage was initiated in a DM channel without the client having the required partial configured.',
+						'If you want PaginatedMessage to work in DM channels then make sure you start your client with "CHANNEL" added to "client.options.partials".',
+						'Furthermore if you are using message based commands (as opposed to application commands) then you will also need to add the "DIRECT_MESSAGE" intent to "client.options.intents"',
+						'If you do not want to be alerted about this in the future then you can disable this warning by setting "PaginatedMessage.emitPartialDMChannelWarning" to "false", or use "setEmitPartialDMChannelWarning(false)" before calling "run".'
+					].join('\n'),
+					{
+						type: 'PaginatedMessageRunsInNonpartialDMChannel',
+						code: 'PAGINATED_MESSAGE_RUNS_IN_NON_PARTIAL_DM_CHANNEL'
+					}
+				);
+				this.hasEmittedPartialDMChannelWarning = true;
 			}
 
-			await this.resolvePagesOnRun();
+			await safelyReplyToInteraction({
+				messageOrInteraction,
+				interactionEditReplyContent: this.#thisMazeWasNotMeantForYouContent,
+				interactionReplyContent: { ...this.#thisMazeWasNotMeantForYouContent, ephemeral: true },
+				componentUpdateContent: this.#thisMazeWasNotMeantForYouContent,
+				messageMethod: 'reply',
+				messageMethodContent: this.#thisMazeWasNotMeantForYouContent
```

Comment:
Thats a lot of duplications... 😓

Reply:
Better than duplicating the whole method thrice :(